### PR TITLE
feat: add interval multiplication

### PR DESCRIPTION
**Pull request by arkanovicz**
Add multiplication of an interval by an integral number `multiply_interval(x,y)`

+ x can be i8, i16, i32 or i64
+ y can be interval_day or interval_year

### CODE REVIEW

File: extensions/functions_datetime.yaml [hunk 1]
```diff
@@ -218,6 +218,58 @@ scalar_functions:
           - name: y
             value: interval_day
         return: timestamp
+  -
+    name: "multiply_interval"
```
Comment (EpsilonPrime) [inline extensions/functions_datetime.yaml:222]: Should this be just another multiply specialization as add is above?

File: extensions/functions_datetime.yaml [hunk 2]
```diff
@@ -218,6 +218,58 @@ scalar_functions:
...
+        return: interval_day
+      - args:
+          - name: x
+            value: i16
+          - name: y
+            value: interval_day
+        return: interval_day
+      - args:
+          - name: x
+            value: i32
```
Comment (EpsilonPrime) [inline extensions/functions_datetime.yaml:239]: Postgres supports double as well.

**Review by EpsilonPrime — COMMENTED** [review]
Postgres additionally supports dividing an interval by a double.

Given the commutative nature of multiply we could consider supporting the reverse order for the arguments but I don't see support for that in any existing documentation so it would likely be unused.

File: extensions/functions_datetime.yaml (hunk 1 above)
Comment (westonpace) [reply extensions/functions_datetime.yaml:222]: +1.  It seems a little odd to have `add`, `subtract`, and then `multiply_interval` instead of `multiply`.

Comment (arkanovicz) [reply extensions/functions_datetime.yaml]: Ok, operator renamed.

File: extensions/functions_datetime.yaml (hunk 2 above)
Comment (arkanovicz) [reply extensions/functions_datetime.yaml:239]: It is apparently the only engine around supporting this. Is it a sufficient reason to add it?

It does so by approximating a month to 30 days whenever necessary...

Comment (EpsilonPrime) [reply extensions/functions_datetime.yaml:239]: Hmm, agreed, that's not an ideal behavior to have lying around.

**Review by EpsilonPrime — COMMENTED** [review]
Looks good to me.  @westonpace Would you like a second pass at this one?

**Review by EpsilonPrime — APPROVED** [review]

File: extensions/functions_datetime.yaml (hunk 2 above)
Comment (westonpace) [reply extensions/functions_datetime.yaml:239]: Yikes, thanks for investigating :)

**Pull request merged**